crates/compression/src/ports.rs

```diff
@@ -18,20 +18,46 @@ use fuel_core_types::{
 /// must be committed atomically, after which block height must be incremented.
 pub trait TemporalRegistry<T> {
     /// Reads a value from the registry at its current height.
-    fn read_registry(&self, key: RegistryKey) -> anyhow::Result<T>;
+    fn read_registry(&self, key: &RegistryKey) -> anyhow::Result<T>;
 
     /// Reads a value from the registry at its current height.
-    fn write_registry(&mut self, key: RegistryKey, value: &T) -> anyhow::Result<()>;
+    fn write_registry(&mut self, key: &RegistryKey, value: &T) -> anyhow::Result<()>;
 
     /// Lookup registry key by the value.
     fn registry_index_lookup(&self, value: &T) -> anyhow::Result<Option<RegistryKey>>;
 }
 
+impl<D, T> TemporalRegistry<T> for &mut D
+where
+    D: TemporalRegistry<T>,
```

Comment:
Why?

Reply:
It allows to use `&mut Database` as an input to `CompressCtx` instead of `Database`